motionplan/planManager.go

```diff
@@ -103,12 +104,21 @@ func (pm *planManager) PlanSingleWaypoint(ctx context.Context,
 		}
 		numSteps := PathStepCount(seedPos, goalPos, pathStepSize)
 
+		// augment the worldstate to allow safe movement with blend radius
+		var modifiedWS *referenceframe.WorldState
+		if worldState != nil {
+			// we only allow a 1 mm blend radius
+			if modifiedWS, err = referenceframe.BufferedWorldstate(worldState, 1); err != nil {
```

Comment:
I don't think this is the right answer here.

For one, operating against a different data structure with changed values compared to what's passed in will make debugging harder.

Secondly, it's not clear how this will affect future types of geometries, notably meshes.

Thirdly and most importantly, this only alters the WorldState geometries, not the self-collision geometries or collisions with other things in the frame system.

I think the correct answer here may be to alter the signature for `geometry.CollidesWith` to take a buffer value and only collide if the separation is less than that amount.

This is what every `CollidesWith` function is already doing internally with the `CollisionBuffer` const, this would just allow us to change from the default value.

Then in Motion, this would require only a change to the collision constraint.

Broadly speaking, we do not want to have changes in our generic motion planning framework that affect everything, but exist only to patch over idiosyncrasies in one specific arm driver.